tests/rptest/tests/data_migrations_api_test.py

```diff
@@ -1,120 +0,0 @@
-# Copyright 2020 Redpanda Data, Inc.
-#
-# Use of this software is governed by the Business Source License
-# included in the file licenses/BSL.md
-#
-# As of the Change Date specified in that file, in accordance with
-# the Business Source License, use of this software will be governed
-# by the Apache License, Version 2.0
-import random
-
-from rptest.services.admin import Admin, MigrationAction
-from rptest.services.admin import OutboundDataMigration, InboundDataMigration, NamespacedTopic, InboundTopic
-
-from rptest.services.cluster import cluster
-from ducktape.utils.util import wait_until
-from rptest.tests.redpanda_test import RedpandaTest
-from ducktape.tests.test import TestContext
-from rptest.clients.types import TopicSpec
-
-
-class DataMigrationsApiTest(RedpandaTest):
-    def __init__(self, test_context: TestContext):
-        super().__init__(test_context=test_context)
-        self.admin = Admin(self.redpanda)
-
-    def get_migrations_map(self, node=None):
-        migrations = self.admin.list_data_migrations(node).json()
-        return {migration["id"]: migration for migration in migrations}
-
-    def wait_for_migration_state(self, id: int, state: str):
-        def migration_in_state():
-            for n in self.redpanda.nodes:
-                migrations = self.get_migrations_map(n)
-                if id not in migrations or migrations[id]["state"] != state:
-                    return False
-            return True
-
-        wait_until(
-            migration_in_state,
-            timeout_sec=90,
-            backoff_sec=1,
-            err_msg=f"Failed waiting for migration {id} to reach {state} state"
-        )
-
-    def create_and_wait(self, migration: InboundDataMigration
-                        | OutboundDataMigration):
-        reply = self.admin.create_data_migration(migration).json()
-        self.logger.info(f"create migration reply: {reply}")
-
-        def migration_is_present(id: int):
-            for n in self.redpanda.nodes:
-                m = self.get_migrations_map(n)
-                if id not in m:
-                    return False
-            return True
-
-        migration_id = reply["id"]
-        wait_until(
-            lambda: migration_is_present(migration_id), 30, 2,
-            f"Expected migration with id {migration_id} is not present")
-        return migration_id
-
-    @cluster(num_nodes=3)
```

Comment:
Is the test still useful, should we keep it as @ok_to_fail or @ignore?
I guess we have the git history if we need to revive it...